syntax = "proto3";
import "NodeStats.proto";
// The greeting service definition.
service CoordinatorRPCService {
  // Sends a greeting
  rpc RegisterNode (RegisterNodeRequest) returns (RegisterNodeReply) {}

  rpc UnregisterNode (UnregisterNodeRequest) returns (UnregisterNodeReply) {}

  rpc RegisterPhysicalStream (RegisterPhysicalStreamRequest) returns (RegisterPhysicalStreamReply) {}

  rpc UnregisterPhysicalStream (UnregisterPhysicalStreamRequest) returns (UnregisterPhysicalStreamReply) {}

  rpc RegisterLogicalStream (RegisterLogicalStreamRequest) returns (RegisterLogicalStreamReply) {}

  rpc UnregisterLogicalStream (UnregisterLogicalStreamRequest) returns (UnregisterLogicalStreamReply) {}

  rpc AddParent (AddParentRequest) returns (AddParentReply) {}

  rpc RemoveParent (RemoveParentRequest) returns (RemoveParentReply) {}

}



message RegisterNodeRequest {
  string address = 1;
  uint64 grpcPort = 2;
  uint64 dataPort = 3;
  uint32 numberOfCpus = 4;
  NES.NodeStats nodeProperties = 5;
  int32 type = 6;
}

message RegisterNodeReply {
  uint64 id = 1;
}

message UnregisterNodeRequest {
  uint64 id = 1;
}

message UnregisterNodeReply {
  bool success = 1;
}

message RegisterPhysicalStreamRequest {
  uint64 id = 1;
  string sourceType = 2;
  string sourceConf = 3;
  int64 sourceFrequency = 4;
  int64 numberOfBuffersToProduce = 5;
  string physicalStreamName = 6;
  string logicalStreamName = 7;
}

message RegisterPhysicalStreamReply {
  bool success = 1;
}

message UnregisterPhysicalStreamRequest {
  uint64 id = 1;
  string physicalStreamName = 2;
  string logicalStreamName = 3;
}

message UnregisterPhysicalStreamReply {
  bool success = 1;
}

message RegisterLogicalStreamRequest {
  uint64 id = 1;
  string streamName = 2;
  string streamSchema = 3;
}

message RegisterLogicalStreamReply {
  bool success = 1;
}

message UnregisterLogicalStreamRequest {
  uint64 id = 1;
  string streamName = 2;
}

message UnregisterLogicalStreamReply {
  bool success = 1;
}

message AddParentRequest {
  uint64 childId = 1;
  uint64 parentId = 2;
}

message AddParentReply {
  bool success = 1;
}

message RemoveParentRequest {
  uint64 childId = 1;
  uint64  parentId = 2;
}

message RemoveParentReply {
  bool success = 1;
}